plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java

```diff
@@ -101,15 +102,26 @@ public AzureBlobStore(RepositoryMetadata metadata, AzureStorageService service,
         final Map<String, AzureStorageSettings> prevSettings = this.service.refreshAndClearCache(emptyMap());
         final Map<String, AzureStorageSettings> newSettings = AzureStorageSettings.overrideLocationMode(prevSettings, this.locationMode);
         this.service.refreshAndClearCache(newSettings);
-        this.getMetricsCollector = (requestMethod) -> {
-            if (requestMethod.equalsIgnoreCase("HEAD")) {
+        this.getMetricsCollector = (httpURLConnection) -> {
+            if (httpURLConnection.getRequestMethod().equalsIgnoreCase("HEAD")) {
                 stats.headOperations.incrementAndGet();
                 return;
             }
 
             stats.getOperations.incrementAndGet();
```

Comment:
Semi-related to this PR since you did it in the new collector, let's add assertions in spots like this one to make sure, we're actually seeing a `GET` here :) Can probably do the same for the list collector also.